server/src/test/java/io/crate/expression/scalar/HasSchemaPrivilegeFunctionTest.java

```diff
@@ -66,22 +81,37 @@ public void test_at_least_one_arg_is_null_returns_null() {
 
     @Test
     public void test_throws_error_when_user_is_not_found() {
-        sqlExpressions = new SqlExpressions(tableSources, null, null, null);
-
-        Asserts.assertThrowsMatches(
-            () -> assertEvaluate("has_schema_privilege('not_existing_user', 'pg_catalog', ' USAGE')", null),
-            IllegalArgumentException.class,
-            "User not_existing_user does not exist"
-        );
+        assertThatThrownBy(
+            () -> assertEvaluate("has_schema_privilege('not_existing_user', 'pg_catalog', ' USAGE')", null))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("User not_existing_user does not exist");
     }
 
     @Test
     public void test_throws_error_when_invalid_privilege() {
-        Asserts.assertThrowsMatches(
-            () -> assertEvaluate("has_schema_privilege('test', 'pg_catalog', ' USAGE, CREATE, SELECT')", null),
-            IllegalArgumentException.class,
-            "Unrecognized privilege type: select"
-        );
+        assertThatThrownBy(
+            () -> assertEvaluate("has_schema_privilege('test', 'pg_catalog', ' USAGE, CREATE, SELECT')", null))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Unrecognized privilege type: select");
+    }
+
+    @Test
+    public void test_throws_error_when_user_is_not_super_user_checking_for_other_user() {
+        sqlExpressions = new SqlExpressions(tableSources, null, TEST_USER, List.of(TEST_SUPER_USER1));
+        assertThatThrownBy(
+            () -> assertEvaluate("has_schema_privilege('testSuperUser', 'pg_catalog', ' USAGE, CREATE, SELECT')", null))
+            .isExactlyInstanceOf(MissingPrivilegeException.class)
+            .hasMessage("Missing privilege for user 'test'");
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void test_throws_error_when_user_is_not_super_user_checking_for_other_user_for_compiled() {
+        assertThatThrownBy(
+            () -> compileScalar("has_schema_privilege('testSuperUser', name, 'USAGE')", "test", () -> List.of(TEST_USER, TEST_SUPER_USER1))
+                .evaluate(txnCtx, sqlExpressions.nodeCtx, Literal.of("testSuperUser"), Literal.of("sys"), Literal.of("USAGE")))
+            .isExactlyInstanceOf(MissingPrivilegeException.class)
+            .hasMessage("Missing privilege for user 'test'");
```

Comment:
Why isn't `assertCompile` working here? Afaik the exception is correctly thrown on compilation so evaluation shouldn't be needed to test the assertion.

Reply:
I was confused with the whole functionality, I have reverted these changes and validate privileges during compilation and not evaluation of compiled version.